### ISSUE TITLE
Problems with Custom Main Menu!

### ISSUE THREAD

**Issue by wizboss1019** [The-Fireplace/In-Game-Account-Switcher#2]
I am having trouble getting In-Game Account Switcher to work with Custom Main Menu. 

Here is the file for my custom menu:

{
    "images":
    {
        "title":
        {
            "image" : "minecraft:textures/gui/title/title.png",
            "posX" : -150,
            "posY" : 10,
            "width" : 305,
            "height" : 128,
            "alignment" : "top_center"
        }
    },

```
"buttons":
{
    "singleplayer":
    {
        "text" : "menu.singleplayer",
        "posX" : -100,
        "posY" : 78,
        "width" : 200,
        "height" : 20,
        "action" : 
        {
            "type" : "openGui",
            "gui" : "singleplayer"
        }
    },

    "multiplayer":
    {
        "text" : "menu.multiplayer",
        "posX" : -100,
        "posY" : 102,
        "width" : 98,
        "height" : 20,
        "action" : 
        {
            "type" : "openGui",
            "gui" : "multiplayer"
        }
    },

    "Kingdoms Server":
    {
        "text" : "Join the Server",
        "posX" : 2,
        "posY" : 102,
        "width" : 98,
        "height" : 20,
        "action" : 
        {
            "type" : "connectToServer",
            "ip" : "<IP NOT FOUND>"
        }
    },
```

 =========================================  
        "accounts":
        {
            "text" : "",
            "posX" : 110,
            "posY" : 162,
            "width" : 20,
            "height" : 20,
            "wrapped Button" : 20,
            "texture" : "minecraft:textures/gui/cmmbutton.png",
            "pressSound" : "",
            "tooltip" : "You are currently logged in as #username#.",
            "action" : 
            {
                "type" : "openGui",
                "gui" : "accswitcher"
            }
        },
  ==========================================
        "mods":
        {
            "text" : "Mods",
            "posX" : -100,
            "posY" : 126,
            "width" : 200,
            "height" : 20,
            "action" : 
            {
                "type" : "openGui",
                "gui" : "mods"
            }
        },

```
    "options":
    {
        "text" : "menu.options",
        "posX" : -100,
        "posY" : 162,
        "width" : 98,
        "height" : 20,
        "action" : 
        {
            "type" : "openGui",
            "gui" : "options"
        }
    },

    "quit":
    {
        "text" : "menu.quit",
        "posX" : 2,
        "posY" : 162,
        "width" : 98,
        "height" : 20,
        "action" : 
        {
            "type" : "quit"
        }
    },

    "language":
    {
        "text" : "",
        "posX" : -124,
        "posY" : 162,
        "width" : 20,
        "height" : 20,
        "action" : 
        {
            "type" : "openGui",
            "gui" : "languages"
        }
    },

    "refresh":
    {
        "text" : "",
        "posX" : 225,
        "posY" : -50,
        "width" : 20,
        "height" : 20,
        "texture" : "custommainmenu:textures/gui/buttons.png",
        "action" : 
        {
            "type" : "refresh"
        }
    }
},

"texts":
{
    "mojang":
    {
        "text" : "Copyright Mojang AB. Do not distribute!",
        "posX" : -197,
        "posY" : -10,
        "color" : -1,
        "alignment" : "bottom_right"
    },

    "changelog":
    {
        "text" : "
```

Version 1.0.5
    +Storyline Beginning!
    +HQM Re-integrated

Version 1.0.4
    +Re-Auth Button
    +More TC Tools & Weapons

Version 1.0.3
    +New Cosmetic Mods
    +New Utility Mods

Version 1.0.2
    +New Menu Background
    +New Menu Format
    -Removed HQM

Version 1.0.1
    +Fixed Recipie Conflicts
    +Performance Boosted!
    +HQM Development
    +New Logo

Version 1.0.0
    +Many Mods
    +Custom Main Menu",
            "posX" : 6,
            "posY" : -245,
            "color" : -10,
            "alignment" : "bottom_left"
        },

```
    "Welcome":
    {
        "text" : "Welcome #username#!",
        "posX" : -50,
        "posY" : 140,
        "color" : -200,
        "hoverColor" : -200,
        "alignment" : "top_center"
    },

    "Date & Time":
    {
        "text" : "It is currently #time# on #date#.",
        "posX" : -90,
        "posY" : 300,
        "color" : -200,
        "hoverColor" :-200,
        "alignment" : "top_center"
    },

    "fml":
    {
        "text" : "",
        "posX" : -175,
        "posY" : -36,
        "color" : -1,
        "alignment" : "top_right"
    }
},

"other":
{
    "splash-text":
    {
        "posX" : 120,
        "posY" : 120,
        "color" : -256,
        "alignment" : "top_center",
        "texts" : "file:minecraft:texts/splashes.txt"
    },

    "panorama":
    {
        "images" : "minecraft:textures/gui/title/background/panorama_%c.png",
        "animate" : true,
        "animationSpeed" : 1,
        "blur" : true,
        "gradient" : false
    }
}
```

}

I put equal signs on top and bottom of the In-Game Account Switcher (I call it IGAS for short) to show it easier (they aren't there in the config file) only major change I made in it was from _**"wrappedButton"**_ to _**"wrapped Button"**_ The button doesn't show without it. Now the button and hovertip are both there, but clicking the button doesn't do anything. Any help will definitely be appreciated. Thank you. 

(btw, IGAS is listed in the mods folder as _"InGameAccountSwitcher-Forge bp1.7.10(6.1.1.72)"_ . I downloaded it at [)](https://mods.curse.com/mc-mods/minecraft/232676-in-game-account-switcher#t1:other-downloads) )

**Comment by The-Fireplace** [comment]
Interesting. Because wrappedButton is what makes it work when clicked. I'll look in to it.

**Comment by The-Fireplace** [comment]
The texture line should be:
"texture" : "accswitcher:textures/gui/cmmbutton.png",
it should be wrappedButton and not wrapped Button.
And there is no
`{
"type" : "openGui",
"gui" : "accswitcher"
}`
Remove that part.

**Comment by wizboss1019** [comment]
@The-Fireplace Thank you for responding. I now did that. Using the text:

"accounts":
        {
            "text" : "",
            "posX" : 110,
            "posY" : 162,
            "width" : 20,
            "height" : 20,
            "wrappedButton" : 20,
            "texture" : "accswitcher:textures/gui/cmmbutton.png",
            "pressSound" : "",
            "tooltip" : "You are currently logged in as #username#."
        },

This makes the button not show entirely. Changing the "wrappedButton" to "wrapped Button" shows the button, but it won't work.  Any suggestions?

**Comment by The-Fireplace** [comment]
@wizboss1019 idk. I will have to give it a try myself and see if I can figure out the cause of the problem.

**Comment by wizboss1019** [comment]
Ok thanks. If you can get back to me if/when you find and/or fix the problem, that would be great! Thank you for these quick responses. I only wish I was able to respond back that fast.

**Comment by The-Fireplace** [comment]
I just tested, and the following config works without issues:
http://paste.ubuntu.com/23528134/

**Comment by wizboss1019** [comment]
@The-Fireplace  Thank you so much for solving it. I am sorry, I just realized I should have told you that (before this last post of yours fixing it) i had that config I gave you in a sub-menu (I guess that is the best way to explain it.). Using the config you provided in this new post, I changed the sub-menu (I labeled it "playmenu") to the config you gave. This did not show the button still. Changing the config labeled "mainmenu" (which was something else) it now works. Just thought I'd let you know what I thought was the problem, even though you might have found this bug. Again, I thank you so much for fixing it, (or at least pointing the problem out). Have a great day!
